lib/rspec/support/caller_filter.rb

```diff
@@ -19,13 +19,19 @@ class CallerFilter
 
     LIB_REGEX = %r{/lib/rspec/(#{(RSPEC_LIBS + ADDITIONAL_TOP_LEVEL_FILES).join('|')})(\.rb|/)}
 
+    LANGUAGE_SPECIFIC_REGEX = []
+    LANGUAGE_SPECIFIC_REGEX <<  "<internal:" if defined?(RUBY_ENGINE) && RUBY_ENGINE == 'truffleruby'
```

Comment:
If we need to add specific things for truffle ruby we'd prefer to add a truffle ruby flag in our ruby module and use that rather than repeat checks everywhere.

Reply:
It would be safer to anchor the pattern though, so "^<internal:" instead of "<internal:"